src/Ast/AST.php

```diff
@@ -48,20 +48,29 @@ protected static function deref($obj): string
         return $obj === static::SELF || $obj instanceof ResolvedType ? '::' : '->';
     }
 
-    protected static function toPhp($x): string
+    protected static function toPhp($x, &$omitSemicolon = false): string
```

Comment:
It's hard to know without having more context, but is this better than `appendSemiColon = true`? I tend to prefer positive variables instead of negative ones, but there are definitely exceptions to that.

Reply:
I tend to prefer a parameter with a default value to use `false` as the default value; but I admit it's not always clear which is the easiest to read. I'll this this as-is for now.